code/_onclick/item_attack.dm

```diff
@@ -1,12 +1,27 @@
 
 /obj/item/proc/melee_attack_chain(mob/user, atom/target, params)
-	if(pre_attackby(target, user, params))
+	if(!tool_check(user, target) && pre_attackby(target, user, params))
 		// Return 1 in attackby() to prevent afterattack() effects (when safely moving items for example)
 		var/resolved = target.attackby(src, user, params)
 		if(!resolved && target && !QDELETED(src))
 			afterattack(target, user, 1, params) // 1: clicking something Adjacent
 
 
+//Checks if the item can work as a tool, calling the appropriate tool behavior on the target
+/obj/item/proc/tool_check(mob/user, atom/target)
+	if(tool_flags & NONE)
+		return

```

Comment:
This will never evaluate to true, do if(!tool_flags) instead.

Reply:
or `if(tool_flags == NONE)`